File: 30DaysOfCode/day25_runningTimeAndComplexity/runningTimeAndComplexity.rb
N = gets.strip.to_i

(0..N-1).each do |i|
    Num = STDIN.readline.strip.to_i
    flag = true
    
    if (Num < 2)
        print("Not prime\n")
    elsif (Num <= 3)
        print("Prime\n")
    elsif(Num % 2 == 0 or Num % 3 == 0)
        print ("Not prime\n")
    else
        for j in (5..(Math.sqrt(Num) + 1)).step(2)
            if (Num % j == 0)
                print("Not prime\n")
                flag = false
                break
            end
        end
        
        if (flag == true)
            print("Prime\n")
        end
    end
end

File: 30DaysOfCode/day25_runningTimeAndComplexity/runningTimeAndComplexity.pl
use Math::Complex;

$N = <STDIN>;
chomp $N;

for ($i = 0; $i < $N; $i++) {
    $num = <STDIN>;
    chomp $num;
    $flag = 1;
    
    if ($num < 2) {
        print "Not prime\n";
    } elsif ($num <= 3) {
        print "Prime\n";
    } elsif ($num % 2 == 0 || $num % 3 == 0) {
        print "Not prime\n";
    } else {
        for ($j = 5; $j <= sqrt($num); $j += 2) {
            if ($num % $j == 0) {
                print "Not prime\n";
                $flag = 0;
                last;
            }
        }
        
        if ($flag == 1) {
            print "Prime\n";
        }
    }
}
